tephra-core/src/main/java/co/cask/tephra/TransactionManager.java

```diff
@@ -490,13 +490,20 @@ private void restoreSnapshot(TransactionSnapshot snapshot) {
    * This is required for backwards compatibility, when long running transactions were represented
    * with expiration time -1. This can be removed when we stop supporting SnapshotCodec version 1.
    */
-  private Map<Long, InProgressTx> txnBackwardsCompatCheck(Map<Long, InProgressTx> inProgress) {
+  public static Map<Long, InProgressTx> txnBackwardsCompatCheck(int defaultLongTimeout,
+                                                                Map<Long, InProgressTx> inProgress) {
     for (Map.Entry<Long, InProgressTx> entry : inProgress.entrySet()) {
-      if (entry.getValue().getExpiration() < 0) {
-        long writePointer = entry.getKey();
-        long expiration = getTxExpirationFromWritePointer(writePointer, defaultLongTimeout);
+      long writePointer = entry.getKey();
+      long expiration = entry.getValue().getExpiration();
+      // LONG transactions will either have a negative expiration or expiration set to the long timeout
+      // use a fudge factor on the expiration check, since expiraton is set based on system time, not the write pointer
+      if (entry.getValue().getType() == null &&
+          (expiration < 0 ||
+              (getTxExpirationFromWritePointer(writePointer, defaultLongTimeout) - expiration < 10000))) {
```

Comment:
I'm not sure it should be either.  If we really want to expose a way to tweak this without a code change, we can make it configurable.  But I don't think it should be documented.  This entire method should be removed as soon as possible -- as soon as we can drop support for the old codec.